internal/bminventory/inventory_test.go

```diff
@@ -6338,12 +6368,94 @@ var _ = Describe("infraEnvs", func() {
 				})
 				Expect(reply).Should(BeAssignableToTypeOf(installer.NewRegisterInfraEnvCreated()))
 				actual := reply.(*installer.RegisterInfraEnvCreated)
-				var err error
 				i, err = bm.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *actual.Payload.ID})
 				Expect(err).ToNot(HaveOccurred())
 				err = db.Model(&common.InfraEnv{}).Where("id = ?", i.ID).Update("generated_at", strfmt.DateTime(time.Now().AddDate(0, 0, -1))).Error
 				Expect(err).ToNot(HaveOccurred())
 			})
+			jsonEncodeStringArray := func(array []string) string {
+				b, e := json.Marshal(&array)
+				Expect(e).ToNot(HaveOccurred())
+				return string(b)
+			}
+			It("Update DiscoveryKernelArguments", func() {
+				By("Do not update DiscoveryKernelArguments -> DiscoveryKernelArguments empty", func() {
+					mockInfraEnvUpdateSuccess()
+					Expect(i.DiscoveryKernelArguments).To(BeNil())
+					reply := bm.UpdateInfraEnv(ctx, installer.UpdateInfraEnvParams{
+						InfraEnvID: *i.ID,
+						InfraEnvUpdateParams: &models.InfraEnvUpdateParams{
+							AdditionalNtpSources: swag.String("1.1.1.1"),
+						},
+					})
+					Expect(reply).To(BeAssignableToTypeOf(installer.NewUpdateInfraEnvCreated()))
+					i, err = bm.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *i.ID})
+					Expect(err).ToNot(HaveOccurred())
+					Expect(i.DiscoveryKernelArguments).To(BeNil())
+				})
+				kp1 := models.KernelArguments{
+					"p1",
+					"p2",
+				}
+
+				By("Update DiscoveryKernelArguments. Expected to be set in db", func() {
+					mockInfraEnvUpdateSuccess()
+					reply := bm.UpdateInfraEnv(ctx, installer.UpdateInfraEnvParams{
+						InfraEnvID: *i.ID,
+						InfraEnvUpdateParams: &models.InfraEnvUpdateParams{
+							DiscoveryKernelArguments: kp1,
+						},
+					})
+					Expect(reply).To(BeAssignableToTypeOf(installer.NewUpdateInfraEnvCreated()))
+					i, err = bm.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *i.ID})
+					Expect(err).ToNot(HaveOccurred())
+					Expect(swag.StringValue(i.DiscoveryKernelArguments)).To(Equal(jsonEncodeStringArray(kp1)))
+				})
+				By("Do not update DiscoveryKernelArguments -> DiscoveryKernelArguments with previous value", func() {
```

Comment:
`It("does not change existing kernel arguments when none are sent in an update"`